components/motor/roboclaw/roboclaw.go

```diff
@@ -318,7 +319,40 @@ func (m *roboclawMotor) GoTo(ctx context.Context, rpm, positionRevolutions float
 }
 
 func (m *roboclawMotor) SetRPM(ctx context.Context, rpm float64, extra map[string]interface{}) error {
-	return motor.NewSetRPMUnsupportedError(m.Name().ShortName())
+	if m.conf.TicksPerRotation == 0 {
+		if rpm > maxRPM {
+			rpm = maxRPM
+		} else if rpm < -1*maxRPM {
+			rpm = -1 * maxRPM
+		}
+		powerPct := rpm / maxRPM
+		m.logger.CInfo(ctx, "speed is an estimation based on a max RPM 250. For increased accuracy, connect encoders")
+		err := m.SetPower(ctx, powerPct, extra)
+		if err != nil {
+			return errors.Wrap(err, "error in SetRPM`")
+		}
+		return nil
+	}
+
+	_, done := m.opMgr.New(ctx)
+	defer done()
+
+	ticksPerSecond := int32((rpm * float64(m.conf.TicksPerRotation)) / 60)
+
+	var err error
+
+	switch m.conf.Channel {
+	case 1:
+		err = m.conn.SpeedDistanceM1(m.addr, ticksPerSecond, uint32(math.Inf(int(rpm))), true)
+	case 2:
+		err = m.conn.SpeedDistanceM2(m.addr, ticksPerSecond, uint32(math.Inf(int(rpm))), true)
+	default:
+		return m.conf.wrongChannelError()
+	}
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
To summarize these 4 lines: if the error is not nil, return the error, and if it is nil, return nil (equivalent to returning the error). Why don't we just put return statements on lines 346 and 348, similar to the one on line 350, and get rid of this part? It seems unnecessarily complicated, but perhaps you've got a plan to use the extra complexity in the near future and it'll make more sense then.

Reply:
yeah good point. this is copied from another section of code so I will fix it in both